### PR TITLE
reshesh cache update error

### LINKED ISSUE #5013: MITx Online Integration: dashboard-up-to-date error message
When a learner's enrollments and grades are out of date, the following error message is displayed:

> Sorry, edx.org is not responding as expected, so some of the information on this page may not be up to date.
>
> Please refresh the browser, or check back later.

Now that we're adding MITx Online as an alternative backend, we need to update this:

- [ ] Update the frontend to display a more generic error message:
  > Sorry, we're unable to update your dashboard at this time, so some of the information on this page may not be up to date.
  >
  > Please refresh the browser, or check back later.
- [ ] Update the `are_all_caches_fresh` to return `False` only if the caches for backends the learner has social auth for are out of date. If the learner doesn't have social auth for edx or mitx online, that should be treated as `True`

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #5013

#### What's this PR do?
- Update the frontend to display a more generic error message
- Update the are_all_caches_fresh to return False only if the caches for backends the learner has social auth for are out of date. If the learner doesn't have social auth for edx or mitx online, that should be treated as True

#### How should this be manually tested?

Create a user that has course enrollments in both courseware backends, and make sure the dashboard loads with no error.
Then make the one of the backends not reachable, and refresh the dashboard, you should see a message at the top saying that the data might not be fresh.

### CODE REVIEW

File: backends/utils.py [hunk 1]
```diff
@@ -56,3 +61,20 @@ def update_email(user_profile_edx, user):
     """
     user.email = user_profile_edx.get('email')
     user.save()
+
+
+def has_social_auth(user, provider):
```
Comment (rhysyngsun) [inline backends/utils.py:66]: Could you add a test for this?

File: dashboard/api_edx_cache.py [hunk 2]
```diff
@@ -133,7 +137,7 @@ def is_cache_fresh(cls, user, cache_type):
         Returns:
             bool
         """
-        if cache_type not in cls.SUPPORTED_CACHES:
+        if cache_type not in cls.SUPPORTED_CACHES + cls.MITXONLINE_SUPPORTED_CACHES:
```
Comment (rhysyngsun) [inline dashboard/api_edx_cache.py:140]: Could you update the tests for the modifications in this module?

**Review by rhysyngsun — CHANGES_REQUESTED** [review]

**Comment by annagav** [comment]
@rhysyngsun the tests should be there now. Thank you

**Review by rhysyngsun — APPROVED** [review]
LGTM

**Pull request merged**